packages/devextreme/js/__internal/scheduler/m_utils_time_zone.ts

```diff
@@ -55,17 +57,66 @@ const getDaylightOffset = (startDate, endDate) => new Date(startDate).getTimezon
 
 const getDaylightOffsetInMs = (startDate, endDate) => getDaylightOffset(startDate, endDate) * toMs('minute');
 
-const calculateTimezoneByValue = (timezone, date = new Date()) => {
-  // NOTE: This check could be removed. We don't support numerical timezones
-  if (typeof timezone === 'string') {
-    const dateUtc = createUTCDate(date);
-    return timeZoneDataUtils.getTimeZoneOffsetById(timezone, dateUtc.getTime());
+const calculateTimezoneByValue = (timeZone, date = new Date()): number | undefined => {
+  if (!timeZone) {
+    return undefined;
+  }
+
+  let offset;
+  try {
+    offset = getStringOffset(timeZone, date);
+  } catch (e) {
+    return undefined;
+  }
+
+  if (offset === 'GMT') {
+    return 0;
   }
-  return timezone;
+
+  const isMinus = offset.substring(3, 4) === '-';
+  const hours = offset.substring(4, 6);
+  const minutes = offset.substring(7, 9);
+
+  const result = parseInt(hours, 10) + parseInt(minutes, 10) / 60;
+  return isMinus ? -result : result;
+};
+
+// 'GMT±XX:YY' or 'GMT' format
+const getStringOffset = (timeZone, date = new Date()): string => {
+  const dateTimeFormat = new Intl.DateTimeFormat('en-US', {
+    timeZone,
+    timeZoneName: 'longOffset',
+  } as any);
+
+  return dateTimeFormat.formatToParts(date)
+    .filter((part) => part.type === 'timeZoneName')
+    .map((p) => p.value)[0];
+};
+
+const getOffsetNamePart = (offset: string): string => {
+  if (offset === 'GMT') {
+    const GMTNamePart = `${offset} +00:00`;
+    return `${GMTNamePart}`;
+  }
+  return offset.replace('GMT', 'GMT ');
+};
+
+const getTimezoneTitle = (timeZone, date = new Date()): string => {
+  const tzNamePart = timeZone.replaceAll('/', ' - ').replaceAll('_', ' ');
+  const offset = getStringOffset(timeZone, date);
```

Comment:
Looks like we need a fallback here if the ```getStringOffset``` will raise an exception / returns undefined (see comment about getting rid of the ```try/catch``` block)